packages/syft/src/syft/service/worker/worker_service.py

```diff
@@ -240,19 +241,19 @@ def stop(
         if isinstance(workers, DockerWorker):
             workers = [workers]
 
-        client = docker.from_env()
-        for w in workers:
-            result = self.stash.delete_by_uid(context.credentials, uid=w.id)
+        with contextlib.closing(docker.from_env()) as client:
+            for w in workers:
+                result = self.stash.delete_by_uid(context.credentials, uid=w.id)
 
-            if result.is_err():
-                return SyftError(message=f"Failed to stop workers {result.err()}")
+                if result.is_err():
+                    return SyftError(message=f"Failed to stop workers {result.err()}")
 
-            # stop container
-            try:
-                client.containers.list(filters={"id": w.container_id})[0].stop()
-                # also prune here?
-            except Exception as e:
-                # we dont throw an error here because apparently the container was already killed
-                print(f"Failed to kill container {e}")
+                # stop container
+                try:
+                    client.containers.list(filters={"id": w.container_id})[0].stop()
+                    # also prune here?
+                except Exception as e:
+                    # we dont throw an error here because apparently the container was already killed
+                    print(f"Failed to kill container {e}")
```

Comment:
@rasswanth-s I noticed for this part, if deleting a worker fails, the function will return immediately, without trying to delete the remaining workers. Should we attempt to delete all workers first, then return a `SyftError` listing the ones that fails if any?

Also after stopping the containers with `stop()` should we remove them too?